lib/logstash/inputs/snmp.rb

```diff
@@ -71,10 +74,19 @@ def register
       retries = host["retries"] || 2
       timeout = host["timeout"] || 1000
 
+      host_details = host_name.match(HOST_REGEX)
+      raise(LogStash::ConfigurationError, "invalid format for host option '#{host_name}'") unless host_details
+      raise(LogStash::ConfigurationError, "only the udp protocol is supported for now") unless host_details[:host_protocol].to_s =~ /udp/i
+
       definition = {
```

Comment:
Have you considered making a Definition class? It paves the way for better handling of `definition[:client].get(definition[:get]`, something like `definition = Definition.new(oid_root_skip, host_name, community, version, retries, timeout, mib, get, walk)`, with `host_name.match(HOST_REGEX)` done in the constructor?
Then `@client_definitions.each {|definition| definition.update(result)`

Reply:
good suggestion. since this is fairly simple for now I'd defer such refactor a bit and wait until I add tcp support and v3 protocol etc. these feature will probably clarify how this should be better structured.